components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/CertificateBasedTokenBinder.java

```diff
@@ -188,14 +192,46 @@ private String generateCnfHashValue(HttpServletRequest request) {
         }
     }
 
-    private X509Certificate parseCertificate(String content) throws CertificateException {
+    /**
+     * Return Certificate for give Certificate Content.
+     *
+     * @param content   Certificate Content
+     * @return X509Certificate X.509 certificate after decoding the certificate content.
+     * @throws CertificateException Certificate Exception.
+     */
+    private X509Certificate parseCertificate(String content) throws CertificateException, UnsupportedEncodingException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Trying to parse the client certificate: " + content);
```

Comment:
do we need to log certifiate now?ince we know the reason.

Reply:
Removed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2438/commits/15b47635c7cf4d8d020c5551c03e246d9d38e3a2